### ISSUE TITLE
Can't connect to the server on mobile devices

### ISSUE THREAD

**Issue by mramonlopez** [feixuwu/UEWebsocket#47]
Hi, I´m using this plugin in my game with no problems running in editor and standalone mode. But when I run my game in physical mobile devices (iOS or Android), the socket doesn't connect to remote server neither throws any error.

Probably It's my fault, but I need some help to known where to look.

Thanks in advance!

**Comment by feixuwu** [comment]
Would u please post the server address?
And when connect fail on mobile device, is there any error message?

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: M. Ramon Lopez <notifications@github.com>
发送时间: Friday, May 31, 2019 10:35:25 PM
收件人: feixuwu/UEWebsocket
抄送: Subscribed
主题: [feixuwu/UEWebsocket] Can't connect to the server on mobile devices (#47)


Hi, I´m using this plugin in my game with no problems running in editor and standalone mode. But when I run my game in physical mobile devices (iOS or Android), the socket doesn't connect to remote server neither throws any error.

Probably It's my fault, but I need some help to known where to look.

Thanks in advance!

**Comment by mramonlopez** [comment]
Hi feixuwu,

I have found this message in my Logcat output:

06-04 17:21:25.442 25787 25811 D UE4     : [2019.06.04-15.21.25:442][  0]LogInit:  websocket: using generated PEM file: '/storage/emulated/0/Android/data/com.onikami.soccer/files/ca-bundle.pem'
read: unexpected EOF!

Do I need add some *.pem file to package?

**Comment by mramonlopez** [comment]
Hi again,

I have deployed a test server to:

ws://chat--socket-example.herokuapp.com

On connect you can send this text 

        {"type":"ROOM_REQUEST", "payload":{"userName": "chatdude"}}

to get some extra responses.

**Comment by feixuwu** [comment]
Hi, I use the demo project and replace the url with your server, the connect is fine, did u device network is enable and can connect to the server?

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: M. Ramon Lopez <notifications@github.com>
发送时间: Wednesday, June 5, 2019 12:42:47 AM
收件人: feixuwu/UEWebsocket
抄送: xufeiwu; Comment
主题: Re: [feixuwu/UEWebsocket] Can't connect to the server on mobile devices (#47)


Hi again,

I have deployed a test server to:

ws://chat--socket-example.herokuapp.com

On connect you can send this text

    {"type":"ROOM_REQUEST", "payload":{"userName": "chatdude"}}


to get some extra responses.

―

**Comment by mramonlopez** [comment]
> Hi, I use the demo project and replace the url with your server, the connect is fine

Have you tested it on a real device?

> did u device network is enable and can connect to the server?

I have a Cocos2D-x version of my game (prototype) using websocket (libwebsocket) connecting to the same server from the same devices and works fine. 

I have added "android.permission.INTERNET" to Android Platform configuration, by the way.

**Comment by mramonlopez** [comment]
After digging in "adb logcat" console out, I have found this:

```
06-12 15:15:40.705 15887 15912 D UE4     : LogPluginManager: Mounting plugin WebSocket
06-12 15:15:40.931 15887 15912 D UE4     : [2019.06.12-13.15.40:931][  0]LogShaderLibrary: Display: Failed to load Shared Shader Library: WebSocket and no native library supported.
06-12 15:15:40.931 15887 15912 D UE4     : [2019.06.12-13.15.40:931][  0]LogShaderLibrary: Display: Failed to load Shared Shader Library: WebSocket_SC and no native library supported.
06-12 15:15:40.931 15887 15912 D UE4     : [2019.06.12-13.15.40:931][  0]LogShaderLibrary: Display: Failed to load Shared Shader Library: WebSocket and no native library supported.
06-12 15:15:40.931 15887 15912 D UE4     : [2019.06.12-13.15.40:931][  0]LogShaderLibrary: Display: Failed to load Shared Shader Library: WebSocket_SC and no native library supported.
```

**Comment by feixuwu** [comment]
yes, I tested on real device, I upload the demo apk here, u can test it:
https://www.dropbox.com/s/5pt17bfebb0ffpu/UEWebsocket-armv7-es2.apk?dl=0

**Comment by feixuwu** [comment]
from your logcat error, it seems like u did not include the libwebsocket.so to the final apk?

**Comment by mramonlopez** [comment]
I found it!!!!!

This code

```
void UNetworkController::sendRoomRequest() {
    UROOM_REQUEST* message = NewObject<UROOM_REQUEST>();
    
    message->type = "ROOM_REQUEST";
    (...)
    FString msgString;
    UWebSocketBlueprintLibrary::ObjectToJson(message, msgString);
    ws_->SendText(msgString);
}
```

sends this correct message to my server from the desktop build
```
{
	"type": "ROOM_REQUEST",
	"payload":
	{
		"userInfo":
		{
			"ID": "unreal-local",
			"name": "Unreal Local",
			"shield":
			{
				"shape": 0,
				"pattern": 2,
				"colors": [ 11, 13, 11 ]
			},
			"roster": [
				{
					"character": 0
				},
				{
					"character": 1
				}
			]
		}
	}
}
```

and this wrong message from mobile version

```
{
	"Type": "ROOM_REQUEST",
	"payload":
	{
		"userInfo":
		{
			"ID": "unreal-local",
			"Name": "Unreal Local",
			"shield":
			{
				"shape": 1,
				"Pattern": 1,
				"Colors": [ 3, 2, 10 ]
			},
			"roster": [
				{
					"Character": 0
				},
				{
					"Character": 1
				}
			]
		}
	}
}
```

 UWebSocketBlueprintLibrary::ObjectToJson is capitalizing  (!!!!) "Type", "Name", "Pattern", "Colors" and "Character"
